apps/sms/views/inbox/js/inbox.js

```diff
@@ -697,106 +632,73 @@ var InboxView = {
-    // the body with the draft content's first string.
-    if (draft && draft.timestamp >= record.timestamp) {
-      body = draft.content.find((content) => typeof content === 'string');
-      type = draft.type;
-    }
+    var labelId = '';
 
-    node.dataset.lastMessageType = type;
+    bodyHTML = Template.escape(bodyHTML || '');
 
-    if (draft) {
-      // Set the "draft" visual indication
-      node.classList.add('draft');
 
+    classList.push('threadlist-item');
+
+    if (draft) {
       if (isDraft) {
-        node.dataset.draftId = draft.id;
-        node.classList.add('is-draft');
-        iconLabel = 'is-draft';
+        labelId = 'is-draft';
       } else {
-        node.classList.add('has-draft');
-        iconLabel = 'has-draft';
+        labelId = 'has-draft';
       }
-    } else {
-      // remove it
-      node.classList.remove('draft', 'has-draft', 'is-draft');
     }
 
     if (record.unreadCount > 0) {
-      node.classList.add('unread');
-      iconLabel = 'unread-thread';
-    } else {
-      node.classList.remove('unread');
-    }
-
-    node.querySelector('.js-conversation-body').textContent = body || '';
-
-    var stateIndicator = node.querySelector('.js-conversation-state');
-    if (iconLabel) {
-      document.l10n.setAttributes(stateIndicator, iconLabel);
-    } else {
-      stateIndicator.removeAttribute('data-l10n-id');
+      labelId = 'unread-thread';
     }
-  },
 
-  createThread: function inbox_createThread(record) {
-    // Create DOM element
-    var li = document.createElement('li');
-    var timestamp = +record.timestamp;
-    var id = record.id;
-    var thread = Threads.get(id);
-    // A "new message" draft
-    var isDraft = thread.isDraft;
-    // An existing conversation's draft.
-    var draft = thread.getDraft();
+    var labelPromise = labelId ?
+      document.l10n.formatValue(labelId) : Promise.resolve('');
 
-    li.id = 'thread-' + id;
-    li.dataset.threadId = id;
-    li.dataset.time = timestamp;
-    li.classList.add('threadlist-item');
-
-    // Render markup with thread data
-    li.innerHTML = this.tmpl.thread.interpolate({
+    return Promise.all([
+      Utils.getHeaderDate(daySection),
+      labelPromise,
+      mmsPromise
+    ]).then((results) => ({
+      type: type,
+      draftId: isDraft ? draft.id : '',
       hash: isDraft ? '#/composer' : '#/thread?id=' + id,
       mode: isDraft ? 'drafts' : 'threads',
       id: isDraft ? draft.id : id,
-      timestamp: String(timestamp)
-    }, {
-      safe: ['id']
-    });
-
-    this.updateConversationContent(li, record);
-
-    TimeHeaders.update(li.querySelector('time'));
-
-    return li;
+      title: participants.join(' , ') || '',
+      bodyHTML: bodyHTML,
+      timestamp: String(timestamp),
+      timeTextContent: Utils.getFormattedHour(timestamp),
+      daySection: daySection,
+      dayTextContent: results[0],
+      iconLabel: results[1],
+      status: labelId,
+      mmsTextContent: results[2]
+    }));
   },
```

Comment:
nit: I think a lot of the patch in bug 1206727 was reverted here, can you look closer when you'll rebase ?